src/pages/RepoPage/CoverageTab/subroute/FileExplorer/FlagMultiSelect.jsx

```diff
@@ -0,0 +1,98 @@
+import eq from 'lodash/eq'
+import isUndefined from 'lodash/isUndefined'
+import PropTypes from 'prop-types'
+import { useState } from 'react'
+
+import { useLocationParams } from 'services/navigation'
+import { useRepoBackfilled, useRepoFlagsSelect } from 'services/repo'
+import { useFlags } from 'shared/featureFlags'
+import Icon from 'ui/Icon'
+import MultiSelect from 'ui/MultiSelect'
+
+const defaultQueryParams = {
+  // without this search here, when updating the params it fills it in for some reason :thinking:
+  search: '',
+  flags: [],
+}
+
+function FlagMultiSelect() {
+  const { params, updateParams } = useLocationParams(defaultQueryParams)
+  const [selectedFlags, setSelectedFlags] = useState(params?.flags)
+  const [flagSearch, setFlagSearch] = useState(null)
+
+  const { data: repoBackfilledData } = useRepoBackfilled()
+
+  const isTimeScaleEnabled = !!repoBackfilledData?.isTimeScaleEnabled
+  const flagsMeasurementsActive = !!repoBackfilledData?.flagsMeasurementsActive
+  const noFlagsPresent = eq(repoBackfilledData?.flagsCount, 0)
+
+  const { coverageTabFlagMultiSelect } = useFlags({
+    coverageTabFlagMultiSelect: false,
+  })
+
+  const {
+    data: flagsData,
+    isLoading: flagsIsLoading,
+    hasNextPage: flagsHasNextPage,
+    fetchNextPage: flagsFetchNextPage,
+  } = useRepoFlagsSelect({
+    filters: { term: flagSearch },
+    options: {
+      suspense: false,
+      enabled:
+        !!coverageTabFlagMultiSelect ||
+        (flagsMeasurementsActive && !noFlagsPresent && isTimeScaleEnabled),
+    },
+  })
+
+  if (!coverageTabFlagMultiSelect || noFlagsPresent) {
+    return null
+  }
+
+  const flagNames = new Set()
+  if (flagsMeasurementsActive) {
+    params?.flags?.forEach((flag) => flagNames.add(flag))
+
+    if (!isUndefined(flagsData)) {
+      flagsData?.forEach((flag) => flagNames.add(flag?.name))
```

Comment:
I believe it's actually required to have the selected items in the list which in this instance is always being added, so that downshift can look through list of items and find see if the selected item appears in the item list.

Reply:
The short and simple, selected values need to be in the list so this is correct behaviour*